### PR TITLE
BLE: fix num <-> asci passkey conversions error

### PR DESCRIPTION
**Pull request by pan-**

### CODE REVIEW

**Comment by pan-** [comment]
@paul-szczepanek-arm Please review; this was needed to compile the file.

**Comment by paul-szczepanek-arm** [comment]
I've already made some of those changes, I'll apply the rest myself to avoid resolving conflicts

File: features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp [hunk 1]
```diff
@@ -370,7 +380,7 @@ class GenericSecurityManager : public SecurityManager,
     }
 
     virtual ble_error_t passkeyEntered(connection_handle_t connection, Passkey_t passkey) {
-        return pal.passkey_request_reply(connection, passkey);
+        return pal.passkey_request_reply(connection, PasskeyAsci::to_num(passkey));
```
Comment (paul-szczepanek-arm) [inline features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp]: no, that passkey is the asci version

Comment (pan-) [reply features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp]: according to the compiler and the code it is the num version:  https://github.com/paul-szczepanek-arm/mbed-os/blob/palsm/features/FEATURE_BLE/ble/pal/PalSecurityManager.h#L458

Comment (paul-szczepanek-arm) [reply features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp]: sorry, I was thinking of the    passkeyEntered(Gap::Handle_t handle, Passkey_t passkey)

Comment (paul-szczepanek-arm) [reply features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp]: no, that is right

Comment (pan-) [reply features/FEATURE_BLE/source/generic/GenericSecurityManager.cpp]: I run compilation to ensure that typewise it's fine. That's why I made this PR.

**Pull request closed without merging**